src/models/components/fully_connected.py

```diff
@@ -0,0 +1,15 @@
+import torch
+import torch.nn as nn
+
+
+class FullyConnected(nn.Module):
+    def __init__(self, dim_in: int, dim_out: int, activation: nn.Module = None):
```

Comment:
Typeヒントは`Callable` です。あとはデフォルト値は`ReLU()`にしておくと便利かと。
```suggestion
    def __init__(self, dim_in: int, dim_out: int, activation: Callable[[Tensor], Tensor] = torch.relu):
```

Reply:
修正しました https://github.com/MLShukai/PrimitiveAMI/pull/97/commits/e7baa40a0ea14638db0dafe381c2730965d3691f   
デフォルトはNone=恒等写像が自然かとも思いましたが

Reply:
恒等写像をNoneにした時はもはや`nn.Linear`と全く等価なので実装する必要性が薄いかと思います

Reply:
@myxyy なるほど、モノイドについてはよくわかっていませんが、リファクタリングや再設計時に参考にさせていただきたいです。ていうか良い性質はどのようなものなのでしょう？

Reply:
今回のケースではモノイドの話はあまり関係ないですが、"何もしない"操作は他の概念（今回だとrelu）に依存せずに存在するためデフォルトの値として適切と考えられます（数学だと**自然に定まる**とか表現される）  
今回の場合の良い性質の具体例はちょっと思いつくのが難しいですが、**余談**に書いているように恒等操作や単位元といった"何もしない"操作を特別視するメリットは確かにあるはずで、そのベストプラクティスを踏襲した形です  
率直な話では線形層とReLU関数というのは本来何の関係もないはずで、そこでReLUがデフォルト値として特別視されるのは気持ち悪い感覚があります

### 余談
モノイドの単位元は再帰的関数における基底ケースの引数として考えることが多いです  
例えば文字列$`x_1x_2\dots x_n`$は列の結合操作$`+`$に対してモノイドを成し、単位元は空列$`\varepsilon`$とした場合について考えます  
$(abc+def)+ghi=abc+(def+ghi)$や$`\varepsilon+abcd=abcd+\varepsilon=abcd`$が成り立つような世界ですね
このとき文字列のマージソートを考えると
```math
\begin{align}
{\rm mergesort}(\varepsilon) &= \varepsilon\\
{\rm mergesort}(s_1 + s_2)  &= {\rm merge}({\rm mergesort}(s_1), {\rm mergesort}(s_2))
\end{align}
```
のような擬似コードとなりこの$`{\rm mergesort}(\varepsilon) = \varepsilon`$が基底ケースです  
単位元は様々なシチュエーションで活用するため特別な名前を付けておく等で識別可能とすると便利です  
PyTorchでも[恒等写像の層](https://pytorch.org/docs/stable/generated/torch.nn.Identity.html#torch.nn.Identity)が定義されているようですね

Reply:
@myxyy 確かに reluを特別視するのは気持ちが悪いですね。その感覚は非常に重要かと思われます。次回からはそのような実装を行う方針でいきましょう。